### PR TITLE
Save model updates

### PR DESCRIPTION
**Pull request by ezequielc**
@alex-deaconu Please review. Thanks

### CODE REVIEW

**Comment by ezequielc** [comment]
@RV-Adrian Could you also confirm it works as expected: 
https://apps-stage-0.risevision.com 
Thanks

**Comment by alex-deaconu** [comment]
@ezequielc one thing I've noticed is that after pressing Publish the Save button is enabled. I think it should be disabled. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu I checked the code for publishing Presentations and it actually doesn't update the Presentation content, it only sets the status to Published of the Presentation in the server. So the Save button should still be enabled as there are unsaved changes in the Editor.

But I don't think this is clear to the users. We should have a conversation with @RV-Adrian to improve that. I think Restore is also not very clear and I remember users complaining about it. Thoughts? Thanks

**Comment by alex-deaconu** [comment]
@ezequielc to confirm, this is happening even if I don't make any changes to the Presentation. I just open a Revised Presentation and press Publish. In this case there are no changes in the Editor (other than the Published flag). I think this may be confusing for the user as it leads them into a Save & Publish loop. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu Ok, I will review this case. Thanks

**Comment by RV-Adrian** [comment]
@ezequielc @alex-deaconu Right now it sounds like a technical issue? (Save enabled after Publish, even with no changes present)
Let me know if not and we'll discuss.

File: web/scripts/editor/controllers/ctr-workspace.js [hunk 1]
```diff
@@ -21,7 +21,18 @@ angular.module('risevision.editor.controllers')
             _initializing = false;
           });
         } else {
-          $scope.hasUnsavedChanges = true;
+          var newObject = angular.copy(newValue);
+          var oldObject = angular.copy(oldValue);
+          var ignoreFields = ['revisionStatusName', 'changeDate',
```
Comment (ezequielc) [inline web/scripts/editor/controllers/ctr-workspace.js]: @alex-deaconu Sorry, I squashed the update. I improved the detection of changes to ignore fields that are not related to the content itself. So now it should not enable Save after publishing. 
It also fixes the issue we mentioned in the previous PR, that Save was being enabled after opening Presentation Properties. Please review. Thanks

Comment (alex-deaconu) [inline web/scripts/editor/controllers/ctr-workspace.js]: @ezequielc I see. To clarify, ignoring `backgroundStyle` and `backgroundScaleToFit` resolves the issues with opening/closing the settings?

Wondering how efficient this code is however in terms of performance (copy object & compare could slow down the digest process). 

Otherwise, looks good. Thanks.

Comment (ezequielc) [inline web/scripts/editor/controllers/ctr-workspace.js]: @alex-deaconu Yes, `backgroundStyle` and `backgroundScaleToFit` fix that issue.

Regarding performance, I share the same concern. I updated the code to avoid checking again when `hasUnsavedChanges` is already set: https://github.com/Rise-Vision/rise-vision-apps/pull/186/files#diff-ca5a6d5167138827bf78a94b1f30d6a4R19
As most of the time the Presentation will have unsaved changes, it will significantly reduce processing.  

Thanks!

Comment (alex-deaconu) [inline web/scripts/editor/controllers/ctr-workspace.js]: @ezequielc good idea. Should work well this way! Thanks for looking into it.

**Comment by alex-deaconu** [comment]
@ezequielc 👍
